src/gui/gui-buffer.c

```diff
@@ -2353,7 +2353,7 @@ gui_buffer_set_input_prompt (struct t_gui_buffer *buffer,
 void
 gui_buffer_set_input (struct t_gui_buffer *buffer, const char *input)
 {
-    if (!buffer || (string_strcmp (buffer->input_buffer, input) == 0))
```

Comment:
I considered moving the condition around the `gui_buffer_undo_snap` and `gui_input_replace_input` lines below, but there doesn't seem to be any point really. An undo point will only be added if the input actually changes.